pkg/storage/pod.go

```diff
@@ -130,7 +130,11 @@ func (s *podStorage) Store(newPods []PodMetricsPoint) {
 								"timestamp", newContainer.Timestamp)
 						}
 					}
+				} else if lastContainer, found := lastPod[newContainer.Name]; found && newContainer.StartTime.After(lastContainer.Timestamp) || !found {
+					prevContainers[newContainer.Name] = ContainerMetricsPoint{Name: newContainer.Name, MetricsPoint: MetricsPoint{StartTime: newContainer.StartTime, Timestamp: newContainer.StartTime}}
 				}
+			} else {
+				prevContainers[newContainer.Name] = ContainerMetricsPoint{Name: newContainer.Name, MetricsPoint: MetricsPoint{StartTime: newContainer.StartTime, Timestamp: newContainer.StartTime}}
```

Comment:
If MS has restarted and it will populate it's cache with very old pods. There is no sense in exposing average CPU usage from container that has run for multiple hours. Metrics Server should report usage from most recent data.

 Let's make assumption that MS metric report window is equal frequency of scrape period (not always true as cAdvisor computes metrics every 10-20s). As so we should consider using start time only if difference between start time and metric timestamp is smaller then metric scrape frequency.